src/spec.rs

```diff
@@ -14,6 +15,52 @@ pub(crate) const ZIP64_CENTRAL_DIRECTORY_END_LOCATOR_SIGNATURE: u32 = 0x07064b50
 pub const ZIP64_BYTES_THR: u64 = u32::MAX as u64;
 pub const ZIP64_ENTRY_THR: usize = u16::MAX as usize;
 
+#[derive(Copy, Clone, Debug)]
+#[repr(packed)]
+pub struct CDEBlock {
+    pub disk_number: u16,
+    pub disk_with_central_directory: u16,
+    pub number_of_files_on_this_disk: u16,
+    pub number_of_files: u16,
+    pub central_directory_size: u32,
+    pub central_directory_offset: u32,
+    pub zip_file_comment_length: u16,
+}
+
+impl CDEBlock {
+    const SIG_LEN: usize = mem::size_of::<u32>();
```

Comment:
I'd instead favor `size_of_val(&ZIP64_CENTRAL_DIRECTORY_END_SIGNATURE)` so that you only have to make the type explicit in one place.

Reply:
If you inline this expression where it's still needed after implementing the signature+block struct, you'll be able to use `size_of_val` and probably still get the performance of a `const`.

Reply:
I've decided to use `Vec` instead of trying to get consts working.